src/spell.c

```diff
@@ -160,72 +158,105 @@ char ilet;
-			Your("gloves seem unaffected.");
-		    } else if (uarmg->oeroded2 < MAX_ERODE) {
-			if (uarmg->greased) {
-			    grease_protect(uarmg, "gloves", &youmonst);
+		case 0:
+			if (already_cursed || rn2(4)) {
+				You_feel("a wrenching sensation.");
+				tele(); /* teleport him */
+			}
+			else {
+				if (bp->blessed) {
+					pline_The("book glows brown.");
+					unbless(bp);
+				}
+				else {
+					pline_The("book glows %s.", NH_BLACK);
+					curse(bp);
+				}
+				bp->bknown = TRUE;
+			}
+			break;
+		case 1:
+			You_feel("threatened.");
+			aggravate();
+			break;
+		case 2:
+			make_blinded(Blinded + rn1(100,250),TRUE);
+			break;
+		case 3:
+			pline_The("book develops a huge set of teeth and bites you!");
+			was_in_use = bp->in_use;
+			bp->in_use = FALSE;
+			losehp(rn1(5, 3), "carnivorous book", KILLED_BY_AN);
+			bp->in_use = was_in_use;
+			break;
+		case 4:
+			You("are frozen in fear!");
+			nomul(rn1(lev*2, lev*2), "frozen by a book");
+			break;
+		case 5:
+			pline_The("book was coated with contact poison!");
+			if (uarmg) {
+				if (uarmg->oerodeproof || !is_corrodeable(uarmg)) {
+				Your("gloves seem unaffected.");
+				} else if (uarmg->oeroded2 < MAX_ERODE) {
+				if (uarmg->greased) {
+					grease_protect(uarmg, "gloves", &youmonst);
+				} else {
+					Your("gloves corrode%s!",
+					 uarmg->oeroded2+1 == MAX_ERODE ?
+					 " completely" : uarmg->oeroded2 ?
+					 " further" : "");
+					uarmg->oeroded2++;
+				}
+				} else
+				Your("gloves %s completely corroded.",
+					 Blind ? "feel" : "look");
+				break;
+			}
+			/* temp disable in_use; death should not destroy the book */
+			bp->in_use = FALSE;
+			losestr(Poison_resistance ? rn1(2,1) : rn1(4,3));
+			dmg = Poison_resistance ? rnd(6) : rnd(10);
+			losehp(dmg, "contact-poisoned spellbook", KILLED_BY_AN);
+			bp->in_use = TRUE;
+			break;
+		case 6:
+			if(Antimagic) {
+				shieldeff(u.ux, u.uy);
+				pline_The("book radiates explosive energy, but you are unharmed!");
 			} else {
-			    Your("gloves corrode%s!",
-				 uarmg->oeroded2+1 == MAX_ERODE ?
-				 " completely" : uarmg->oeroded2 ?
-				 " further" : "");
-			    uarmg->oeroded2++;
+				pline("As you read the book, it radiates explosive energy right at you!");
+				dmg = 2*rnd(10) + 10;
 			}
-		    } else
-			Your("gloves %s completely corroded.",
-			     Blind ? "feel" : "look");
-		    break;
-		}
-		/* temp disable in_use; death should not destroy the book */
-		bp->in_use = FALSE;
-		losestr(Poison_resistance ? rn1(2,1) : rn1(4,3));
-		losehp(rnd(Poison_resistance ? 6 : 10),
-		       "contact-poisoned spellbook", KILLED_BY_AN);
-		bp->in_use = TRUE;
-		break;
-	case 6:
-		if(Antimagic) {
-		    shieldeff(u.ux, u.uy);
-		    pline_The("book %s, but you are unharmed!", explodes);
-		} else {
-		    pline("As you read the book, it %s in your %s!",
-			  explodes, body_part(FACE));
-		    losehp(2*rnd(10)+5, "exploding rune", KILLED_BY_AN);
-		}
+			
+			if (Half_spell_damage) dmg = (dmg+1/2);
```

Comment:
For extra fun, why not also check `u.uvaul_duration` and reduce damage taken again? 😄